scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java

```diff
@@ -1452,6 +1467,21 @@ public Builder setSupportedCipherSuites(String... cipherSuites) {
 			}
 			return setSupportedCipherSuites(suites);
 		}
+		
+		/**
+		 * Sets the signature algorithms supported by the connector.
+		 * <p>
+		 * The connector will use these signature algorithms (in exactly the same
+		 * order) during the DTLS handshake.
+		 * 
+		 * @param supportedSignatureAlgorithms the supported signature algorithms in the order of
+		 *            preference
+		 * @return this builder for command chaining
+		 */
+		public Builder setSupportedSignatureAlgorithms(List<SignatureAndHashAlgorithm> supportedSignatureAlgorithms) {
+			config.supportedSignatureAlgorithms = supportedSignatureAlgorithms;
```

Comment:
For the other list in the configuration `ListUtils.init(List)` is called. That checks for double and wraps the list into a "unmodifiableList". Unfortunately that also requires to add `hashCode` and `equals` in `SignatureAndHashAlgorithm`.

Reply:
I added the `ListUtils.init(List)` call in the `public DtlsConnectorConfig build()` method and I have implemented the `hashCode` and `equals` methods in the `SignatureAndHashAlgorithm` class.